change log level and frequency

lindemeier opened the pull request:
ref https://github.com/mojin-robotics/mbl/issues/153

The ```frame_publisher``` node produces error messages when tf lookup fails. This results in log spam.

This PR changes the log level and frequency to reduce log spam.

Commit:
change log level and frequency

HannesBachter reviewed (changes requested):
http://wiki.ros.org/tf/Overview/Exceptions

Commit:
catch each exception an handle them according to criticality

fmessmer commented:
I don't like the throttle part....
in our usecase (in travis) we don't want any output from that node....and only atf to report failure or sucdess...in case of failure, then it needs to be re-run locally for debugging....where output of this node is to be seen again...

this should rather be fixed by an approach proposes (but not yet working) in https://github.com/mojin-robotics/mojin_atf/pull/6

floweisshardt reviewed (commented):
I'm OK with this solution: we still get error logs but not as frequent as before (we did the same for eband and LTS logs). So this PR is not related to ATF runs.

We can still combine that with https://github.com/mojin-robotics/mojin_atf/pull/6 for supressing all output during an ATF run.

fmessmer reviewed (changes requested):
ok, convinced

just two more remarks:
- why don't we use the same `ROS_X_THROTTLE(T,...)` for all exceptions?
- should we still catch `tf2:TransformException` (or another super class) at the end?
- why `tf2::` and not `tf::` as mentioned in the link by @HannesBachter?

Commit:
consistent log level and catching TransformException

lindemeier commented:
>     * why don't we use the same `ROS_X_THROTTLE(T,...)` for all exceptions?
Just considered different exceptions as more critical. Changed all to the same level, since they all can be considered as "error"


>     * why `tf2::` and not `tf::` as mentioned in the link by @HannesBachter?
the buffer being used for the lookup is tf2 as well as the other components.  tf2 offers more types of exceptions

>     * should we still catch `tf2:TransformException` (or another super class) at the end?
```TransformException``` derives directly from ```std::runtime_error``` and according to [this](http://docs.ros.org/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1Buffer.html#acabbd72cae8f49fb3b6ede3be7e34c55) lookkup only throws these exceptions.
I've just added ```TransformException``` additionally to catch also exceptions that might be added in the future (good point thx).

fmessmer reviewed (approved):
thanks for iterating
